src/json.ts

```diff
@@ -289,6 +290,7 @@ export function isPendingJsonRpcResponse(
  */
 export function assertIsPendingJsonRpcResponse(
   response: unknown,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
Hmm. I'm guessing we didn't anticipate passing classes as variables. This is too bad. Maybe worth loosening this rule a bit to allow for this.

Reply:
I wonder if it's possible to only allow certain names for certain types, i.e., only allow `PascalCase` if the type is a class constructor.

Reply:
From looking at https://typescript-eslint.io/rules/naming-convention/ it appears that these rules can filter by types, but only primitive types are supported at the moment. So it looks like only allowing pascal case for class constructors is not yet possible.

Reply:
This rule has been updated here to allow PascalCase parameters: https://github.com/MetaMask/eslint-config/pull/264